the app will compile error with google mobile ad

### Android framework version

net10.0-android

### Affected platform version

.net 10

### Description

compile error with  `javac.exe error JAVAC0000`

### Steps to Reproduce

1. new dotnet maui
2. config android manifest for  below meta
```
<application android:allowBackup="true" android:icon="@mipmap/appicon" android:roundIcon="@mipmap/appicon_round" android:supportsRtl="true">
	<meta-data
		android:name="com.google.android.gms.ads.APPLICATION_ID"
		android:value="ca-app-pub-xxxxxxxxxxx~xxxxxxxxxxxxx"/>
</application>
```
3. invoke the code 
```
#if ANDROID
using Android.App;
using Android.Nfc;
using Android.Util;
using Google.Android.Gms.Ads;
using Google.Android.Gms.Ads.AdManager;
using Google.Android.Gms.Ads.Initialization;
using Google.Android.Gms.Ads.Interstitial;
using Java.Lang;
using System.Diagnostics;
#endif

namespace SampleApp;

public partial class MainPage : ContentPage
{

	public MainPage()
	{
		InitializeComponent();
#if ANDROID
		Task.Run(() =>
		{
            MobileAds.Initialize(Platform.AppContext!,new InitListener());
        });
#endif
	}

	private void OnCounterClicked(object? sender, EventArgs e)
	{

#if ANDROID
        AdManagerInterstitialAd.Load(
            Platform.AppContext!,
            "/21775744923/example/interstitial",
            new AdManagerAdRequest.Builder().Build(),
            new MyInterstitialAdCallBack());
#endif

    }
}

#if ANDROID
public class InitListener : Java.Lang.Object, Google.Android.Gms.Ads.Initialization.IOnInitializationCompleteListener
{
    public void OnInitializationComplete(IInitializationStatus p0)
    {
    }
}

public class MyInterstitialAdCallBack : AdManagerInterstitialAdLoadCallback 
{
    public override void OnAdLoaded(Java.Lang.Object p0)
    {
        var ad = p0 as AdManagerInterstitialAd;
        ad?.Show(Platform.CurrentActivity!);
    }

}
```

#endif

4. Start to debug

### Did you find any workaround?

No

### Relevant log output

```shell
Build started at 21:36...
1>------ Build started: Project: SampleApp, Configuration: Debug Any CPU ------
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.3 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Activity.Ktx 1.10.1.3 requires Xamarin.AndroidX.Activity (>= 1.10.1.3 && < 1.10.2) but version Xamarin.AndroidX.Activity 1.13.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Collection.Ktx 1.5.0.3 requires Xamarin.AndroidX.Collection.Jvm (>= 1.5.0.3 && < 1.5.1) but version Xamarin.AndroidX.Collection.Jvm 1.6.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.ViewModel.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.ViewModel (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.ViewModel 2.11.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.SavedState.SavedState.Ktx 1.3.1.1 requires Xamarin.AndroidX.SavedState (>= 1.3.1.1 && < 1.3.2) but version Xamarin.AndroidX.SavedState 1.5.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime.Android (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime.Android 2.11.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.11.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Process 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.11.0.1 was resolved.
1>  XAML source generation is enabled (MauiXamlInflator=SourceGen). This generates C# code from XAML at compile time for better performance. To disable, remove the MauiXamlInflator line from your .csproj file (reverts to configuration-based defaults: Runtime in Debug, XamlC in Release).
1>  Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers.
1>  Including assemblies for Hot Reload support
1>  SampleApp -> C:\Users\ericc\Desktop\SampleApp\bin\Debug\net10.0-android\SampleApp.dll
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.3 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Activity.Ktx 1.10.1.3 requires Xamarin.AndroidX.Activity (>= 1.10.1.3 && < 1.10.2) but version Xamarin.AndroidX.Activity 1.13.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Collection.Ktx 1.5.0.3 requires Xamarin.AndroidX.Collection.Jvm (>= 1.5.0.3 && < 1.5.1) but version Xamarin.AndroidX.Collection.Jvm 1.6.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.ViewModel.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.ViewModel (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.ViewModel 2.11.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.SavedState.SavedState.Ktx 1.3.1.1 requires Xamarin.AndroidX.SavedState (>= 1.3.1.1 && < 1.3.2) but version Xamarin.AndroidX.SavedState 1.5.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime.Android (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime.Android 2.11.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.11.0.1 was resolved.
1>C:\Users\ericc\Desktop\SampleApp\SampleApp.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Process 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.11.0.1 was resolved.
1>obj\Debug\net10.0-android\android\src\crc64338477404e88479c\GenericAnimatorListener.java(50,14): javac.exe error JAVAC0000:  警告: [removal] Object 中的 finalize() 已过时, 且标记为待删除
1>obj\Debug\net10.0-android\android\src\crc64338477404e88479c\GenericAnimatorListener.java(50,14): javac.exe error JAVAC0000: 	public void finalize ()
1>obj\Debug\net10.0-android\android\src\crc64338477404e88479c\GenericAnimatorListener.java(50,14): javac.exe error JAVAC0000:
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\InterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000:  错误: 名称冲突: InterstitialAdCallBack 中的 onAdLoaded(Object) 和 AdLoadCallback 中的 onAdLoaded(AdManagerInterstitialAd) 具有相同疑符, 但两者均不覆盖对方
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\InterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 	public void onAdLoaded (java.lang.Object p0)
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\InterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000:
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000:  错误: 名称冲突: MyInterstitialAdCallBack 中的 onAdLoaded(Object) 和 AdLoadCallback 中的 onAdLoaded(AdManagerInterstitialAd) 具有相同疑符, 但两者均不覆盖对方
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 	public void onAdLoaded (java.lang.Object p0)
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 注: 某些输入文件使用或覆盖了已过时的 API。
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 注: 有关详细信息, 请使用 -Xlint:deprecation 重新编译。
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 注: 某些输入文件使用了未经检查或不安全的操作。
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 注: 有关详细信息, 请使用 -Xlint:unchecked 重新编译。
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 2 个错误
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000: 1 个警告
1>obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java(26,14): javac.exe error JAVAC0000:
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: obj\Debug\net10.0-android\android\src\crc64338477404e88479c\GenericAnimatorListener.java:50: 警告: [removal] Object 中的 finalize() 已过时, 且标记为待删除
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 	public void finalize ()
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 	            ^
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\InterstitialAdCallBack.java:26: 错误: 名称冲突: InterstitialAdCallBack 中的 onAdLoaded(Object) 和 AdLoadCallback 中的 onAdLoaded(AdManagerInterstitialAd) 具有相同疑符, 但两者均不覆盖对方
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 	public void onAdLoaded (java.lang.Object p0)
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 	            ^
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: obj\Debug\net10.0-android\android\src\crc64ad5f528fd2cb43c0\MyInterstitialAdCallBack.java:26: 错误: 名称冲突: MyInterstitialAdCallBack 中的 onAdLoaded(Object) 和 AdLoadCallback 中的 onAdLoaded(AdManagerInterstitialAd) 具有相同疑符, 但两者均不覆盖对方
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 	public void onAdLoaded (java.lang.Object p0)
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 	            ^
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 注: 某些输入文件使用或覆盖了已过时的 API。
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 注: 有关详细信息, 请使用 -Xlint:deprecation 重新编译。
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 注: 某些输入文件使用了未经检查或不安全的操作。
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 注: 有关详细信息, 请使用 -Xlint:unchecked 重新编译。
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 2 个错误
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\36.1.69\tools\Xamarin.Android.Javac.targets(161,5): error XAJVC0000: 1 个警告
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build completed at 21:37 and took 39.442 seconds ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========
========== Deploy completed at 21:37 and took 39.442 seconds ==========
```

--- a/config.json
+++ b/config.json
@@ -2730,7 +2730,7 @@
         "groupId": "com.google.android.gms",
         "artifactId": "play-services-ads",
         "version": "25.4.0",
-        "nugetVersion": "125.4.0.1",
+        "nugetVersion": "125.4.0.2",
         "nugetId": "Xamarin.GooglePlayServices.Ads",
         "allowPrereleaseDependencies": true,
         "extraDependencies": "androidx.lifecycle.lifecycle-livedata-core,androidx.lifecycle.lifecycle-runtime",
@@ -2740,7 +2740,7 @@
         "groupId": "com.google.android.gms",
         "artifactId": "play-services-ads-api",
         "version": "25.4.0",
-        "nugetVersion": "125.4.0.1",
+        "nugetVersion": "125.4.0.2",
         "nugetId": "Xamarin.GooglePlayServices.Ads.Api",
         "type": "xbd"
       },

--- a/config.json
+++ b/config.json
@@ -2764,7 +2764,7 @@
         "groupId": "com.google.android.gms",
         "artifactId": "play-services-ads-lite",
         "version": "24.0.0",
-        "nugetVersion": "124.0.0.6",
+        "nugetVersion": "124.0.0.7",
         "nugetId": "Xamarin.GooglePlayServices.Ads.Lite",
         "frozen": true,
         "extraDependencies": "androidx.lifecycle.lifecycle-livedata-core,androidx.lifecycle.lifecycle-runtime",

--- a/source/com.google.android.gms/play-services-ads-api/PublicAPI/PublicAPI.Unshipped.txt
+++ b/source/com.google.android.gms/play-services-ads-api/PublicAPI/PublicAPI.Unshipped.txt
@@ -227,6 +227,17 @@ Google.Android.Gms.Ads.Initialization.IOnInitializationCompleteListener.OnInitia
 Google.Android.Gms.Ads.Initialization.InitializationCompleteEventArgs
 Google.Android.Gms.Ads.Initialization.InitializationCompleteEventArgs.InitializationCompleteEventArgs(Google.Android.Gms.Ads.Initialization.IInitializationStatus! p0) -> void
 Google.Android.Gms.Ads.Initialization.InitializationCompleteEventArgs.P0.get -> Google.Android.Gms.Ads.Initialization.IInitializationStatus!
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpDeepLinkServiceWrapper
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpDeepLinkServiceWrapper.EndSession(Android.Gms.Dynamic.IObjectWrapper! p0, string! p1) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpDeepLinkServiceWrapper.Open(Android.Gms.Dynamic.IObjectWrapper! p0, string! p1, string! p2, Android.OS.Bundle! p3, bool p4, Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpServiceCallback! p5) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpDeepLinkServiceWrapper.Prewarm(Android.Gms.Dynamic.IObjectWrapper! p0, System.Collections.Generic.IList<Android.OS.Bundle!>! p1, Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpPrewarmServiceCallback! p2) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpPrewarmServiceCallback
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpPrewarmServiceCallback.OnError(Android.OS.Bundle! p0) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpPrewarmServiceCallback.OnPrewarmCompleted(Android.OS.Bundle! p0) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpServiceCallback
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpServiceCallback.OnDismissed(Android.OS.Bundle! p0) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpServiceCallback.OnError(Android.OS.Bundle! p0) -> void
+Google.Android.Gms.Ads.Internal.Client.Hsdp.IHsdpServiceCallback.OnShown(Android.OS.Bundle! p0) -> void
 Google.Android.Gms.Ads.Internal.Offline.Buffering.OfflineNotificationPoster
 Google.Android.Gms.Ads.Internal.Offline.Buffering.OfflineNotificationPoster.OfflineNotificationPoster(Android.Content.Context! context, AndroidX.Work.WorkerParameters! params) -> void
 Google.Android.Gms.Ads.Internal.Offline.Buffering.OfflineNotificationPoster.OfflineNotificationPoster(nint javaReference, Android.Runtime.JniHandleOwnership transfer) -> void
@@ -1375,6 +1386,7 @@ virtual Google.Android.Gms.Ads.AdLoader.IsLoading.get -> bool
 virtual Google.Android.Gms.Ads.AdLoader.LoadAd(Google.Android.Gms.Ads.AdManager.AdManagerAdRequest! adManagerAdRequest) -> void
 virtual Google.Android.Gms.Ads.AdLoader.LoadAd(Google.Android.Gms.Ads.AdRequest! adRequest) -> void
 virtual Google.Android.Gms.Ads.AdLoader.LoadAds(Google.Android.Gms.Ads.AdRequest! adRequest, int maxNumberOfAds) -> void
+virtual Google.Android.Gms.Ads.AdManager.AdManagerInterstitialAdLoadCallback.OnAdLoaded(Google.Android.Gms.Ads.AdManager.AdManagerInterstitialAd! p0) -> void
 virtual Google.Android.Gms.Ads.AdRequest.AdString.get -> string?
 virtual Google.Android.Gms.Ads.AdRequest.Builder.Build() -> Google.Android.Gms.Ads.AdRequest!
 virtual Google.Android.Gms.Ads.AdRequest.Builder.Self() -> Google.Android.Gms.Ads.AdRequest.Builder!
@@ -1387,6 +1399,7 @@ virtual Google.Android.Gms.Ads.AdRequest.Keywords.get -> System.Collections.Gene
 virtual Google.Android.Gms.Ads.AdRequest.NeighboringContentUrls.get -> System.Collections.Generic.IList<string!>!
 virtual Google.Android.Gms.Ads.AdRequest.PlacementId.get -> long
 virtual Google.Android.Gms.Ads.AdRequest.RequestAgent.get -> string!
+virtual Google.Android.Gms.Ads.AppOpen.AppOpenAd.AppOpenAdLoadCallback.OnAdLoaded(Google.Android.Gms.Ads.AppOpen.AppOpenAd! p0) -> void
 virtual Google.Android.Gms.Ads.BaseAdView.AdListener.get -> Google.Android.Gms.Ads.AdListener!
 virtual Google.Android.Gms.Ads.BaseAdView.AdListener.set -> void
 virtual Google.Android.Gms.Ads.BaseAdView.AdSize.get -> Google.Android.Gms.Ads.AdSize?
@@ -1412,6 +1425,7 @@ virtual Google.Android.Gms.Ads.FullScreenContentCallback.OnAdDismissedFullScreen
 virtual Google.Android.Gms.Ads.FullScreenContentCallback.OnAdFailedToShowFullScreenContent(Google.Android.Gms.Ads.AdError! p0) -> void
 virtual Google.Android.Gms.Ads.FullScreenContentCallback.OnAdImpression() -> void
 virtual Google.Android.Gms.Ads.FullScreenContentCallback.OnAdShowedFullScreenContent() -> void
+virtual Google.Android.Gms.Ads.Interstitial.InterstitialAdLoadCallback.OnAdLoaded(Google.Android.Gms.Ads.Interstitial.InterstitialAd! p0) -> void
 virtual Google.Android.Gms.Ads.Mediation.Adapter.LoadAppOpenAd(Google.Android.Gms.Ads.Mediation.MediationAppOpenAdConfiguration! p0, Google.Android.Gms.Ads.Mediation.IMediationAdLoadCallback! callback) -> void
 virtual Google.Android.Gms.Ads.Mediation.Adapter.LoadBannerAd(Google.Android.Gms.Ads.Mediation.MediationBannerAdConfiguration! p0, Google.Android.Gms.Ads.Mediation.IMediationAdLoadCallback! callback) -> void
 virtual Google.Android.Gms.Ads.Mediation.Adapter.LoadInterstitialAd(Google.Android.Gms.Ads.Mediation.MediationInterstitialAdConfiguration! p0, Google.Android.Gms.Ads.Mediation.IMediationAdLoadCallback! callback) -> void
@@ -1503,8 +1517,10 @@ virtual Google.Android.Gms.Ads.RequestConfiguration.TagForChildDirectedTreatment
 virtual Google.Android.Gms.Ads.RequestConfiguration.TagForUnderAgeOfConsent.get -> int
 virtual Google.Android.Gms.Ads.RequestConfiguration.TestDeviceIds.get -> System.Collections.Generic.IList<string!>!
 virtual Google.Android.Gms.Ads.RequestConfiguration.ToBuilder() -> Google.Android.Gms.Ads.RequestConfiguration.Builder!
+virtual Google.Android.Gms.Ads.Rewarded.RewardedAdLoadCallback.OnAdLoaded(Google.Android.Gms.Ads.Rewarded.RewardedAd! p0) -> void
 virtual Google.Android.Gms.Ads.Rewarded.ServerSideVerificationOptions.CustomData.get -> string!
 virtual Google.Android.Gms.Ads.Rewarded.ServerSideVerificationOptions.UserId.get -> string!
+virtual Google.Android.Gms.Ads.RewardedInterstitial.RewardedInterstitialAdLoadCallback.OnAdLoaded(Google.Android.Gms.Ads.RewardedInterstitial.RewardedInterstitialAd! p0) -> void
 virtual Google.Android.Gms.Ads.VersionInfo.MajorVersion.get -> int
 virtual Google.Android.Gms.Ads.VersionInfo.MicroVersion.get -> int
 virtual Google.Android.Gms.Ads.VersionInfo.MinorVersion.get -> int

--- a/source/com.google.android.gms/play-services-ads-api/Transforms/Metadata.xml
+++ b/source/com.google.android.gms/play-services-ads-api/Transforms/Metadata.xml
@@ -125,4 +125,63 @@
 		OnAdFailedToLoadWithErrorCode
 	</attr>
 
+	<!--
+	  Java-generics erasure fix for AdLoadCallback<AdT> subclasses (issue #1491).
+
+	  The base class:
+	      public abstract class AdLoadCallback<AdT> {
+	          public void onAdLoaded(AdT ad) { }
+	      }
+	  only declares the erased onAdLoaded(Object) in bytecode, so the binding
+	  generator exposes it as `virtual OnAdLoaded(Java.Lang.Object p0)` on every
+	  concrete subclass (AdManagerInterstitialAdLoadCallback,
+	  InterstitialAdLoadCallback, etc.).
+
+	  When a user overrides that in C#, the generated Java ACW stub emits
+	  `public void onAdLoaded(java.lang.Object p0)`. From javac's point of view
+	  the parent (after generic substitution) declares
+	  `onAdLoaded(AdManagerInterstitialAd)`, and the two methods have the same
+	  erasure but neither overrides the other; javac reports a name-clash and
+	  the app fails to build.
+
+	  Fix: for each concrete subclass, add a specialized onAdLoaded(SpecificAd)
+	  method via add-node. That surfaces a properly-typed
+	  `virtual OnAdLoaded(SpecificAd p0)` in C# with a specialized JNI
+	  registration, so users' ACW stubs emit the specialized signature that
+	  correctly overrides the parent.
+
+	  Note: play-services-ads-lite also ships bindings for the same types
+	  (under the Android.Gms.Ads.* namespace); the equivalent fix lives in
+	  that project's Metadata.xml.
+	-->
+	<add-node path="/api/package[@name='com.google.android.gms.ads.admanager']/class[@name='AdManagerInterstitialAdLoadCallback']">
+		<method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/admanager/AdManagerInterstitialAd;)V" jni-return="V">
+			<parameter name="p0" type="com.google.android.gms.ads.admanager.AdManagerInterstitialAd" not-null="true"></parameter>
+		</method>
+	</add-node>
+
+	<add-node path="/api/package[@name='com.google.android.gms.ads.interstitial']/class[@name='InterstitialAdLoadCallback']">
+		<method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/interstitial/InterstitialAd;)V" jni-return="V">
+			<parameter name="p0" type="com.google.android.gms.ads.interstitial.InterstitialAd" not-null="true"></parameter>
+		</method>
+	</add-node>
+
+	<add-node path="/api/package[@name='com.google.android.gms.ads.rewarded']/class[@name='RewardedAdLoadCallback']">
+		<method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/rewarded/RewardedAd;)V" jni-return="V">
+			<parameter name="p0" type="com.google.android.gms.ads.rewarded.RewardedAd" not-null="true"></parameter>
+		</method>
+	</add-node>
+
+	<add-node path="/api/package[@name='com.google.android.gms.ads.rewardedinterstitial']/class[@name='RewardedInterstitialAdLoadCallback']">
+		<method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/rewardedinterstitial/RewardedInterstitialAd;)V" jni-return="V">
+			<parameter name="p0" type="com.google.android.gms.ads.rewardedinterstitial.RewardedInterstitialAd" not-null="true"></parameter>
+		</method>
+	</add-node>
+
+	<add-node path="/api/package[@name='com.google.android.gms.ads.appopen']/class[@name='AppOpenAd.AppOpenAdLoadCallback']">
+		<method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/appopen/AppOpenAd;)V" jni-return="V">
+			<parameter name="p0" type="com.google.android.gms.ads.appopen.AppOpenAd" not-null="true"></parameter>
+		</method>
+	</add-node>
+
 </metadata>

--- a/source/com.google.android.gms/play-services-ads-lite/PublicAPI/PublicAPI.Unshipped.txt
+++ b/source/com.google.android.gms/play-services-ads-lite/PublicAPI/PublicAPI.Unshipped.txt
@@ -1336,6 +1336,7 @@ virtual Android.Gms.Ads.AdLoader.IsLoading.get -> bool
 virtual Android.Gms.Ads.AdLoader.LoadAd(Android.Gms.Ads.AdManager.AdManagerAdRequest! adManagerAdRequest) -> void
 virtual Android.Gms.Ads.AdLoader.LoadAd(Android.Gms.Ads.AdRequest! adRequest) -> void
 virtual Android.Gms.Ads.AdLoader.LoadAds(Android.Gms.Ads.AdRequest! adRequest, int maxNumberOfAds) -> void
+virtual Android.Gms.Ads.AdManager.AdManagerInterstitialAdLoadCallback.OnAdLoaded(Android.Gms.Ads.AdManager.AdManagerInterstitialAd! p0) -> void
 virtual Android.Gms.Ads.AdRequest.AdString.get -> string?
 virtual Android.Gms.Ads.AdRequest.Builder.Build() -> Android.Gms.Ads.AdRequest!
 virtual Android.Gms.Ads.AdRequest.ContentUrl.get -> string!
@@ -1346,6 +1347,7 @@ virtual Android.Gms.Ads.AdRequest.IsTestDevice(Android.Content.Context! context)
 virtual Android.Gms.Ads.AdRequest.Keywords.get -> System.Collections.Generic.ICollection<string!>!
 virtual Android.Gms.Ads.AdRequest.NeighboringContentUrls.get -> System.Collections.Generic.IList<string!>!
 virtual Android.Gms.Ads.AdRequest.RequestAgent.get -> string!
+virtual Android.Gms.Ads.AppOpen.AppOpenAd.AppOpenAdLoadCallback.OnAdLoaded(Android.Gms.Ads.AppOpen.AppOpenAd! p0) -> void
 virtual Android.Gms.Ads.BaseAdView.AdListener.get -> Android.Gms.Ads.AdListener!
 virtual Android.Gms.Ads.BaseAdView.AdListener.set -> void
 virtual Android.Gms.Ads.BaseAdView.AdSize.get -> Android.Gms.Ads.AdSize?
@@ -1369,6 +1371,7 @@ virtual Android.Gms.Ads.FullScreenContentCallback.OnAdDismissedFullScreenContent
 virtual Android.Gms.Ads.FullScreenContentCallback.OnAdFailedToShowFullScreenContent(Android.Gms.Ads.AdError! p0) -> void
 virtual Android.Gms.Ads.FullScreenContentCallback.OnAdImpression() -> void
 virtual Android.Gms.Ads.FullScreenContentCallback.OnAdShowedFullScreenContent() -> void
+virtual Android.Gms.Ads.Interstitial.InterstitialAdLoadCallback.OnAdLoaded(Android.Gms.Ads.Interstitial.InterstitialAd! p0) -> void
 virtual Android.Gms.Ads.Mediation.Adapter.LoadAppOpenAd(Android.Gms.Ads.Mediation.MediationAppOpenAdConfiguration! p0, Android.Gms.Ads.Mediation.IMediationAdLoadCallback! callback) -> void
 virtual Android.Gms.Ads.Mediation.Adapter.LoadBannerAd(Android.Gms.Ads.Mediation.MediationBannerAdConfiguration! p0, Android.Gms.Ads.Mediation.IMediationAdLoadCallback! callback) -> void
 virtual Android.Gms.Ads.Mediation.Adapter.LoadInterstitialAd(Android.Gms.Ads.Mediation.MediationInterstitialAdConfiguration! p0, Android.Gms.Ads.Mediation.IMediationAdLoadCallback! callback) -> void
@@ -1454,8 +1457,10 @@ virtual Android.Gms.Ads.RequestConfiguration.TagForChildDirectedTreatment.get ->
 virtual Android.Gms.Ads.RequestConfiguration.TagForUnderAgeOfConsent.get -> int
 virtual Android.Gms.Ads.RequestConfiguration.TestDeviceIds.get -> System.Collections.Generic.IList<string!>!
 virtual Android.Gms.Ads.RequestConfiguration.ToBuilder() -> Android.Gms.Ads.RequestConfiguration.Builder!
+virtual Android.Gms.Ads.Rewarded.RewardedAdLoadCallback.OnAdLoaded(Android.Gms.Ads.Rewarded.RewardedAd! p0) -> void
 virtual Android.Gms.Ads.Rewarded.ServerSideVerificationOptions.CustomData.get -> string!
 virtual Android.Gms.Ads.Rewarded.ServerSideVerificationOptions.UserId.get -> string!
+virtual Android.Gms.Ads.RewardedInterstitial.RewardedInterstitialAdLoadCallback.OnAdLoaded(Android.Gms.Ads.RewardedInterstitial.RewardedInterstitialAd! p0) -> void
 virtual Android.Gms.Ads.VersionInfo.MajorVersion.get -> int
 virtual Android.Gms.Ads.VersionInfo.MicroVersion.get -> int
 virtual Android.Gms.Ads.VersionInfo.MinorVersion.get -> int

--- a/source/com.google.android.gms/play-services-ads-lite/Transforms/Metadata.xml
+++ b/source/com.google.android.gms/play-services-ads-lite/Transforms/Metadata.xml
@@ -127,6 +127,62 @@
             <parameter name="bundle" type="android.os.Bundle"></parameter>
         </method>
     </add-node>
+
+    <!--
+      Java-generics erasure fix for AdLoadCallback<AdT> subclasses (issue #1491).
+
+      The base class:
+          public abstract class AdLoadCallback<AdT> {
+              public void onAdLoaded(AdT ad) { }
+          }
+      only declares the erased `onAdLoaded(Object)` in bytecode, so the binding
+      generator exposes it as `virtual OnAdLoaded(Java.Lang.Object p0)` on every
+      concrete subclass (AdManagerInterstitialAdLoadCallback, InterstitialAdLoadCallback,
+      etc.).
+
+      When a user overrides that in C#, the generated Java ACW stub emits
+      `public void onAdLoaded(java.lang.Object p0)`. From javac's point of view
+      the parent (after generic substitution) declares
+      `onAdLoaded(AdManagerInterstitialAd)`, and the two methods have the same
+      erasure but neither overrides the other — javac reports a name-clash and
+      the app fails to build.
+
+      Fix: for each concrete subclass, add a specialized `onAdLoaded(SpecificAd)`
+      method via <add-node>. That surfaces a properly-typed
+      `virtual OnAdLoaded(SpecificAd p0)` in C# with a specialized JNI
+      registration, so users' ACW stubs emit the specialized signature that
+      correctly overrides the parent. This matches how Google's own docs show
+      the callback being overridden in Java.
+    -->
+    <add-node path="/api/package[@name='com.google.android.gms.ads.admanager']/class[@name='AdManagerInterstitialAdLoadCallback']">
+        <method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/admanager/AdManagerInterstitialAd;)V" jni-return="V">
+            <parameter name="p0" type="com.google.android.gms.ads.admanager.AdManagerInterstitialAd" not-null="true"></parameter>
+        </method>
+    </add-node>
+
+    <add-node path="/api/package[@name='com.google.android.gms.ads.interstitial']/class[@name='InterstitialAdLoadCallback']">
+        <method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/interstitial/InterstitialAd;)V" jni-return="V">
+            <parameter name="p0" type="com.google.android.gms.ads.interstitial.InterstitialAd" not-null="true"></parameter>
+        </method>
+    </add-node>
+
+    <add-node path="/api/package[@name='com.google.android.gms.ads.rewarded']/class[@name='RewardedAdLoadCallback']">
+        <method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/rewarded/RewardedAd;)V" jni-return="V">
+            <parameter name="p0" type="com.google.android.gms.ads.rewarded.RewardedAd" not-null="true"></parameter>
+        </method>
+    </add-node>
+
+    <add-node path="/api/package[@name='com.google.android.gms.ads.rewardedinterstitial']/class[@name='RewardedInterstitialAdLoadCallback']">
+        <method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/rewardedinterstitial/RewardedInterstitialAd;)V" jni-return="V">
+            <parameter name="p0" type="com.google.android.gms.ads.rewardedinterstitial.RewardedInterstitialAd" not-null="true"></parameter>
+        </method>
+    </add-node>
+
+    <add-node path="/api/package[@name='com.google.android.gms.ads.appopen']/class[@name='AppOpenAd.AppOpenAdLoadCallback']">
+        <method abstract="false" deprecated="not deprecated" final="false" name="onAdLoaded" native="false" return="void" static="false" synchronized="false" visibility="public" bridge="false" synthetic="false" jni-signature="(Lcom/google/android/gms/ads/appopen/AppOpenAd;)V" jni-return="V">
+            <parameter name="p0" type="com.google.android.gms.ads.appopen.AppOpenAd" not-null="true"></parameter>
+        </method>
+    </add-node>
     
     
     <attr

